python/lsst/ts/standardscripts/auxtel/track_target_and_take_image.py

```diff
@@ -225,3 +225,30 @@ def is_standard_visit(self) -> bool:
                 for item in (self.config.exp_times, self.grating, self.band_filter)
             ]
         )
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state to execute the
```

Comment:
why not put this in the base class (line 214), then add 3 keyword parameters
TCS=True --> for self.atcs.assert_all_enabled()
LATISS=TRue --> self.latiss.assert_all_enabled()
AOS_LOOPS --> `        ataos_corrections = await self.atcs.rem.ataos.evt_correctionEnabled.aget(
            timeout=self.atcs.fast_timeout
        )`

Then you don't need to copy/paste code all over the place?

Reply:
But I agree with you. I am not particularly happy to have the same code duplicated here and there.. I could add `assert_feasibilty` to the `TCS` class but I am on the fence about that.